TestingLowerBounds/ForMathlib/ERealLogExp.lean

```diff
@@ -0,0 +1,233 @@
+  induction' x using EReal.rec with x <;> simp
+
+lemma exp_mono {a b : EReal} (h : a ≤ b) : exp a ≤ exp b := by
+  induction' a using EReal.rec with a
+  · simp
+  · induction' b using EReal.rec with b
+    · simp at h
+    · rw [exp_coe, exp_coe]
+      exact ENNReal.ofReal_le_ofReal (Real.exp_le_exp_of_le (mod_cast h))
+    · simp
+  · rw [top_le_iff] at h
+    simp [h.symm, exp_top, top_le_iff]
+
+lemma exp_neg (x : EReal) : exp (-x) = (exp x)⁻¹ := by
+  induction' x using EReal.rec with x
+  · simp
+  · rw [exp_coe, ← EReal.coe_neg, exp_coe, ← ENNReal.ofReal_inv_of_pos (Real.exp_pos _)]
+    congr
+    rw [Real.exp_neg]
+  · simp
+
+lemma exp_add (x y : EReal) : exp (x + y) = exp x * exp y := by
+  induction' x using EReal.rec with x
+  · simp
+  · induction' y using EReal.rec with y
+    · simp
+    · simp only [← EReal.coe_add, exp_coe]
+      rw [← ENNReal.ofReal_mul (Real.exp_nonneg _), Real.exp_add]
+    · simp only [coe_add_top, exp_top, exp_coe]
+      rw [ENNReal.mul_top]
+      simp [Real.exp_pos]
+  · induction' y using EReal.rec with y
+    · simp
+    · simp only [top_add_coe, exp_top, exp_coe]
+      rw [ENNReal.top_mul]
+      simp [Real.exp_pos]
+    · simp
+
+end Exp
+
+section LogExp
+
+@[simp] lemma log_exp (x : EReal) : log (exp x) = x := by
+  induction' x using EReal.rec with x
+  · simp
+  · rw [exp_coe, log_ofReal, if_neg (not_le.mpr (Real.exp_pos _)), Real.log_exp]
+  · simp
+
+@[simp] lemma exp_log (x : ℝ≥0∞) : exp (log x) = x := by
+  by_cases hx_top : x = ∞
+  · simp [hx_top]
+  by_cases hx_zero : x = 0
+  · simp [hx_zero]
+  have hx_pos : 0 < x.toReal := ENNReal.toReal_pos hx_zero hx_top
+  rw [← ENNReal.ofReal_toReal hx_top, log_ofReal_of_pos hx_pos, exp_coe, Real.exp_log hx_pos]
+
+@[simp]
+lemma log_le_iff {a b : ℝ≥0∞} : log a ≤ log b ↔ a ≤ b := by
+  refine ⟨fun h ↦ ?_, log_mono⟩
+  rw [← exp_log a, ← exp_log b]
+  exact exp_mono h
+
+@[simp]
+lemma exp_le_iff {a b : EReal} : exp a ≤ exp b ↔ a ≤ b := by
+  conv_rhs => rw [← log_exp a, ← log_exp b, log_le_iff]
+
+/-- `EReal.log` and its inverse `Ereal.exp` are an order isomorphism between `ℝ≥0∞` and `EReal`. -/
+noncomputable
+def logOrderIso : ℝ≥0∞ ≃o EReal where
+  toFun := log
+  invFun := exp
+  left_inv x := exp_log x
+  right_inv x := log_exp x
+  map_rel_iff' := by simp only [Equiv.coe_fn_mk, log_le_iff, forall_const]
+
+lemma continuous_log : Continuous log := logOrderIso.continuous
+
+lemma continuous_exp : Continuous exp := logOrderIso.symm.continuous
+
+lemma measurable_log : Measurable log := continuous_log.measurable
+
+lemma measurable_exp : Measurable exp := continuous_exp.measurable
+
+lemma _root_.Measurable.ereal_log {α : Type*} {_ : MeasurableSpace α}
+    {f : α → ℝ≥0∞} (hf : Measurable f) :
+    Measurable fun x ↦ log (f x) := measurable_log.comp hf
+
+lemma _root_.Measurable.ereal_exp {α : Type*} {_ : MeasurableSpace α}
+    {f : α → EReal} (hf : Measurable f) :
+    Measurable fun x ↦ exp (f x) := measurable_exp.comp hf
+
+end LogExp
+
```

Comment:
With `logOrderIso` we basically get the metrizability of `EReal` for free, should we add it here?

```suggestion

instance : TopologicalSpace.MetrizableSpace EReal :=
  logOrderIso.symm.toHomeomorph.embedding.metrizableSpace

```